exts/rag_bge_small_en_v15/src/lib.rs

```diff
@@ -167,8 +167,23 @@ pub extern "C" fn background_main(arg: pg_sys::Datum) {
                 .add_service(EmbeddingGeneratorServer::new(embedder))
                 .serve_with_incoming_shutdown(uds_stream, async {
                     while !BackgroundWorker::sigterm_received() {
-                        sleep(Duration::from_millis(500)).await;
-                    }
+                        unsafe {
+                            let mask = pg_sys::WL_EXIT_ON_PM_DEATH | pg_sys::WL_LATCH_SET;
+                            let my_latch = &mut (*pg_sys::MyProc).procLatch as *mut pg_sys::Latch;
+                            let events = pg_sys::WaitLatch(
+                                my_latch,
+                                mask as i32,
+                                0,
+                                pg_sys::PG_WAIT_EXTENSION,
+                            );
+                            pg_sys::ResetLatch(my_latch);
+
+                            if (events & pg_sys::WL_EXIT_ON_PM_DEATH as i32) != 0 {
+                                // postmaster died
+                                pg_sys::proc_exit(1);
+                            }
+                        }
```

Comment:
I think you can use pgrx's [wait_latch](https://docs.rs/pgrx/latest/pgrx/bgworkers/struct.BackgroundWorker.html#method.wait_latch) function for this